core/src/impl/Kokkos_Profiling.cpp

```diff
@@ -202,6 +245,93 @@ void markEvent(const std::string& eventName) {
   }
 }
 
+SpaceHandle make_space_handle(const char* space_name){
+  SpaceHandle handle;
+  handle.name = space_name;
+  return handle;
+}
+
+} // end namespace Profiling
+
+
+namespace Tuning {
+
+static std::unordered_map<size_t, std::unordered_set<size_t>> features_per_context; 
+static std::unordered_set<size_t> active_features;
+static std::unordered_map<size_t,VariableValue> feature_values;
+
+void declareTuningVariable(const std::string& variableName, size_t uniqID, VariableInfo info){
+  if(nullptr != tuningVariableDeclarationCallee){
+    (*tuningVariableDeclarationCallee)(variableName.c_str(), uniqID, info);
```

Comment:
At the very least I need documentation.

What this function does is let a tool know about a tuning parameter it might have to tune, the one below is letting it know about a context variable. Notably, we're not yet saying the value, just "if you see something with unique Id uniqID referenced in the future, this is everything we know about it"